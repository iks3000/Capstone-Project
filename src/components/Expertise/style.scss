@import '../../assets/styles/_variables.scss';
@import '../../assets/styles/_extends.scss';

.expertise {
    margin: 50px 0;

    &__item {
        display: flex;
        margin-bottom: 30px;

        &:last-child {
            margin-bottom: 0;
        }
    }

    &__title {
        @extend %title;
    }

    &__item-date {
        width: 110px;
        flex: 1 0 auto;
    }

    &__content-title {
        @extend %content-title;
        margin-bottom: 5px;
    }

    &__content-date {
        font: 400 16px/19px $primary-fonts;
        color: $raven;
    }

    &__item-info {
        flex: 1 1 auto;
    }
}