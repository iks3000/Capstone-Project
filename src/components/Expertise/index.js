import { useState } from "react";
import Info from "../../components/Info";
import "./style.scss";

const Expertise = ({ data }) => {
  const [dataSeed] = useState(data);

  return (
    <div className="expertise" id="expertiseComponent">
      <h2 className="expertise__title">Experience</h2>
      <ul>
        {dataSeed.map((dataExperience, index) => (
          <li className="expertise__item" key={index}>
            <div className="expertise__item-date">
              <h3 className="expertise__content-title">
                {dataExperience.info.company}
              </h3>
              <p className="expertise__content-date">{dataExperience.date}</p>
            </div>
            <div className="expertise__item-info">
              <h3 className="expertise__content-title">
                {dataExperience.info.job}
              </h3>
              <Info text={dataExperience.info.description} />
            </div>
          </li>
        ))}
      </ul>
    </div>
  );
};

export default Expertise;
