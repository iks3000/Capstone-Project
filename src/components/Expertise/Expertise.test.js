import React from "react";
import { shallow } from "enzyme";
import Expertise from "./index";

it("should render Expertise component", () => {
  const mockItem = {
    data: [
      {
        date: "2013-2014",
        info: {
          company: "Google",
          job: "Front-end developer / php programmer",
          description: "Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor. Aenean massa. Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Donec quam felis, ultricies nec, pellentesque eu, pretium quis, sem. Nulla consequat massa quis enim. Donec pede justo, fringil",
        },
      },
      {
        date: "2012",
        info: {
          company: "Twitter",
          job: "Web developer",
          description: "Donec pede justo, fringilla vel, aliquet nec, vulputate eget, arcu. In enim justo, rhoncus ut, imperdiet a, venenatis vitae, justo. Nullam dictum felis eu pede mollis pretium. Integer tincidunt. Cras dapibus. Vivamus elementum semper nisi. Aenean vulputate eleifend tellus. Aenean leo ligula, porttitor eu, consequat vitae, eleifend ac, enim",
        },
      },
    ],
  };
  expect(shallow(<Expertise {...mockItem} />)).toMatchSnapshot();
});
