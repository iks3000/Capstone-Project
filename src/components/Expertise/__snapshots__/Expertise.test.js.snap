// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Expertise component 1`] = `
<div
  className="expertise"
  id="expertiseComponent"
>
  <h2
    className="expertise__title"
  >
    Experience
  </h2>
  <ul>
    <li
      className="expertise__item"
      key="0"
    >
      <div
        className="expertise__item-date"
      >
        <h3
          className="expertise__content-title"
        >
          Google
        </h3>
        <p
          className="expertise__content-date"
        >
          2013-2014
        </p>
      </div>
      <div
        className="expertise__item-info"
      >
        <h3
          className="expertise__content-title"
        >
          Front-end developer / php programmer
        </h3>
        <Info
          text="Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor. Aenean massa. Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Donec quam felis, ultricies nec, pellentesque eu, pretium quis, sem. Nulla consequat massa quis enim. Donec pede justo, fringil"
        />
      </div>
    </li>
    <li
      className="expertise__item"
      key="1"
    >
      <div
        className="expertise__item-date"
      >
        <h3
          className="expertise__content-title"
        >
          Twitter
        </h3>
        <p
          className="expertise__content-date"
        >
          2012
        </p>
      </div>
      <div
        className="expertise__item-info"
      >
        <h3
          className="expertise__content-title"
        >
          Web developer
        </h3>
        <Info
          text="Donec pede justo, fringilla vel, aliquet nec, vulputate eget, arcu. In enim justo, rhoncus ut, imperdiet a, venenatis vitae, justo. Nullam dictum felis eu pede mollis pretium. Integer tincidunt. Cras dapibus. Vivamus elementum semper nisi. Aenean vulputate eleifend tellus. Aenean leo ligula, porttitor eu, consequat vitae, eleifend ac, enim"
        />
      </div>
    </li>
  </ul>
</div>
`;
